Sources/NostrSDK/DirectMessageEncrypting.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  DirectMessageEncrypting.swift
+//  
+//
+//  Created by Joel Klabo on 8/10/23.
+//
+
+import Foundation
+import secp256k1
+import CommonCrypto
+import CryptoKit
+
+public enum DirectMessageEncryptingError: Error {
+    case pubkeyParsing
+    case sharedSecretCreation
+    case encryptionError
+    case decryptionError
+    case missingValue
+}
+
+public protocol DirectMessageEncrypting {}
+public extension DirectMessageEncrypting {
+
+    func encrypt(content: String, privateKey: PrivateKey, publicKey: PublicKey) throws -> String {
+
+        guard let sharedSecret = try? getSharedSecret(privateKey: privateKey, recipient: publicKey) else {
+            throw EventCreatingError.invalidInput
+        }
+        
+        let iv = randomBytes(count: 16).bytes
+        let utf8Content = Data(content.utf8).bytes
+        guard let encryptedMessage = AESEncrypt(data: utf8Content, iv: iv, sharedSecret: sharedSecret) else {
+            throw DirectMessageEncryptingError.encryptionError
+        }
+
+        return encodeDMBase64(content: encryptedMessage.bytes, iv: iv)
+    }
+
+    func decrypt(encryptedContent message: String, privateKey: PrivateKey, publicKey: PublicKey) throws -> String {
+        guard let sharedSecret = try? getSharedSecret(privateKey: privateKey, recipient: publicKey) else {
+            throw EventCreatingError.invalidInput
+        }
+
+        let sections = Array(message.split(separator: "?"))
+
+        if sections.count != 2 {
+            throw DirectMessageEncryptingError.decryptionError
+        }
+
+        guard let encryptedContent = sections.first,
+              let encryptedContentData = Data(base64Encoded: String(encryptedContent)) else {
+            throw DirectMessageEncryptingError.decryptionError
+        }
+
+        guard let ivContent = sections.last else {
+            throw DirectMessageEncryptingError.decryptionError
+        }
+
+        let ivContentTrimmed = ivContent.dropFirst(3)
+
+        guard let ivContentData = Data(base64Encoded: String(ivContentTrimmed)) else {
```

Comment:
Suggestion: Since you're throwing the same error for these last three `guard let`s, you could collapse them into one:
```swift
guard let ivContentData = Data(base64Encoded: String(ivContentTrimmed)),
      let decryptedContentData = AESDecrypt(data: encryptedContentData.bytes, iv: ivContentData.bytes, sharedSecret: sharedSecret),
      let decryptedMessage = String(data: decryptedContentData, encoding: .utf8) else {
    throw DirectMessageEncryptingError.decryptionError
}
```